Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp

```diff
@@ -0,0 +1,392 @@
+        m_filterModel->SetFilter(m_searchWidget->GetFilter());
+
+        SetupAssetBrowser();
+    }
+
+    void AssetBrowserTest::TearDownEditorFixtureImpl()
+    {
+        m_modelTesterAssetBrowser.reset();
+        m_modelTesterFilterModel.reset();
+        m_modelTesterTableModel.reset();
+
+        m_tableModel.reset();
+        m_filterModel.reset();
+        m_assetBrowserComponent->Deactivate();
+
+        m_assetBrowserComponent.reset();
+        m_searchWidget.reset();
+    }
+
+    void AssetBrowserTest::AddScanFolder(
+        AZ::s64 folderID, AZStd::string folderPath, AZStd::string displayName, FolderType folderType /*= FolderType::File*/)
+    {
+        AzToolsFramework::AssetDatabase::ScanFolderDatabaseEntry scanFolder = AzToolsFramework::AssetDatabase::ScanFolderDatabaseEntry();
+        scanFolder.m_scanFolderID = folderID;
+        scanFolder.m_scanFolder = folderPath;
+        scanFolder.m_displayName = displayName;
+        scanFolder.m_isRoot = folderType == FolderType::Root;
+        GetRootEntry()->AddScanFolder(scanFolder);
+    }
+
+    AZ::Uuid AssetBrowserTest::CreateSourceEntry(
+        AZ::s64 fileID, AZ::s64 parentFolderID, AZStd::string filename, AssetEntryType sourceType /*= AssetEntryType::Source*/)
+    {
+        AzToolsFramework::AssetDatabase::FileDatabaseEntry entry = AzToolsFramework::AssetDatabase::FileDatabaseEntry();
+        entry.m_scanFolderPK = parentFolderID;
+        entry.m_fileID = fileID;
+        entry.m_fileName = filename;
+        entry.m_isFolder = sourceType == AssetEntryType::Folder;
+        GetRootEntry()->AddFile(entry);
+
+        if (!entry.m_isFolder)
+        {
+            AzToolsFramework::AssetBrowser::SourceWithFileID entrySource = AzToolsFramework::AssetBrowser::SourceWithFileID();
+            entrySource.first = entry.m_fileID;
+            entrySource.second = AzToolsFramework::AssetDatabase::SourceDatabaseEntry();
+            entrySource.second.m_scanFolderPK = parentFolderID;
+            entrySource.second.m_sourceName = filename;
+            entrySource.second.m_sourceID = fileID;
+            entrySource.second.m_sourceGuid = AZ::Uuid::CreateRandom();
+
+            GetRootEntry()->AddSource(entrySource);
+
+            return entrySource.second.m_sourceGuid;
+        }
+
+        return AZ::Uuid::CreateNull();
+    }
+
+    void AssetBrowserTest::CreateProduct(AZ::s64 productID, AZ::Uuid sourceUuid, AZStd::string productName)
+    {
+        AzToolsFramework::AssetBrowser::ProductWithUuid product = AzToolsFramework::AssetBrowser::ProductWithUuid();
+        product.first = sourceUuid;
+        product.second = AzToolsFramework::AssetDatabase::ProductDatabaseEntry();
+        product.second.m_productID = productID;
+
+        product.second.m_subID = aznumeric_cast<AZ::u32>(productID);
+        product.second.m_productName = productName;
+
+        GetRootEntry()->AddProduct(product);
+    }
+
+    void AssetBrowserTest::SetupAssetBrowser()
+    {
+
```

Comment:
uber nit: Please remove this blank line (might be worth running clang-format over this file too)